skyportal/models/telescope.py

```diff
@@ -67,75 +67,83 @@ def observer(self):
         try:
             return self._observer
         except AttributeError:
-            if not self.fixed_location:
+            # as this function tries to return self._observer (if the observer already exists), and it there is no self._observer, we need to create it
+            # it should be here that we initialize it to None if it is missing lon, lat, or if its location is not fixed (i.e, a ground based telescope)
+            if self.lon is None or self.lon == "" or np.isnan(self.lon):
                 return None
-
-            try:
-                tf = timezonefinder.TimezoneFinder(in_memory=True)
-                local_tz = tf.timezone_at(
-                    lng=(self.lon + 180) % 360 - 180, lat=self.lat
-                )
-                elevation = self.elevation
-                if (
-                    self.elevation is None
-                    or self.elevation == ""
-                    or np.isnan(self.elevation)
-                ):
-                    elevation = 0
-
-                self._observer = astroplan.Observer(
-                    longitude=self.lon * u.deg,
-                    latitude=self.lat * u.deg,
-                    elevation=elevation * u.m,
-                    timezone=local_tz,
-                )
-
-            except Exception as e:
-                log(
-                    f'Telescope {self.id} ("{self.name}") cannot calculate an observer: {e}'
-                )
+            if self.lat is None or self.lat == "" or np.isnan(self.lat):
+                return None
+            if self.fixed_location is False or self.fixed_location is None:
                 return None
 
+        try:
+            tf = timezonefinder.TimezoneFinder(in_memory=True)
+            local_tz = tf.timezone_at(lng=(self.lon + 180) % 360 - 180, lat=self.lat)
+            elevation = self.elevation
+            # if elevation is not specified, assume it is 0
+            if (
+                self.elevation is None
+                or self.elevation == ""
+                or np.isnan(self.elevation)
+            ):
+                elevation = 0
+
+            self._observer = astroplan.Observer(
+                longitude=self.lon * u.deg,
+                latitude=self.lat * u.deg,
+                elevation=elevation * u.m,
+                timezone=local_tz,
+            )
+
+        except Exception as e:
+            log(
+                f'Telescope {self.id} ("{self.name}") cannot calculate an observer: {e}'
+            )
+            return None
+
         return self._observer
 
+    # in all the methods below that use this self.observer, I checked first if it is None, rather than having an if statement on the "time" value first.
+    # anyway, if the observer is None, we won't need the "time" value, so it seems better to check the observer first.
+    # It will avoid unnecessary calculations if the observer is None.
```

Comment:
Just so you know, you can also add comments on github (like reviewer comments but on your own code). 
That may be easier that adding code comments and then cleaning them up, but it is totally up to you either way.